cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -117,7 +141,8 @@ public void setConnectionConfig(@Nullable ConnectionConfig connectionConfig) {
   public void tryConnect(ConnectionConfig connectionConfig, PrintStream output, boolean debug) throws Exception {
     try {
       AccessToken accessToken = acquireAccessToken(clientConfig, connectionConfig, output, debug);
-      checkConnection(clientConfig, connectionConfig, accessToken);
+      connectionConfig = clientConfig.getConnectionConfig();
```

Comment:
I agree we need refactoring here, but `checkConnection` doesn't take an incomplete `ClientConfig` though, it's intended to be the current `ClientConfig` that has various timeout, etc. props that we want override with the new connection.